frappe/model/db_query.py

```diff
@@ -779,7 +779,7 @@ def build_match_conditions(self, as_condition=True) -> str | list:
 
 			# share is an OR condition, if there is a role permission
 			if not only_if_shared and self.shared and conditions:
-				conditions = f"({conditions}) or ({self.get_share_condition()})"
```

Comment:
The output of all this is already bracket-ed

https://github.com/frappe/frappe/blob/c14379ce5c316714890e0473bdf7c385a70baff0/frappe/model/db_query.py#L535-L537

Can you share example of permission setup + shared document where this breaks? 


If it's specific to contact query the brackets should be added there or in build_match_condition?

Reply:
Hi @ankush thanks for your quick response,

as far as I understand, the upper part in lines 535 to 537 are not executed when `DatabaseQuery(doctype).build_match_conditions(as_condition=as_condition)` is called. Therefore, the match_conditions are not added in brackets.
Our setup to get to the problematic sql statement is the following:

1. Create two contacts which are not linked to the current user
2. Current user should not have special rights as user permissions (i.e. not System Manager)
3. Add a User Permission (DocType User Permissions) to the current user referencing one of the above contacts
4. For the other contact add a DocShare for the current user
5. Go to e.g. Quotation and checkout the Contact Person field
6. The `mcond` I get in `contact_query` looks sth. like this:
`((((ifnull(`tabContact`.`name`, '')='' or `tabContact`.`name` in ('Leo'))))) or (`tabContact`.name in ('Bruno B', 'Leo'))`
the brackets are not around the whole statement which leads to the issue in the sql WHERE clause.

Thanks for your review and best regards,
Leo